js-envs/src/main/scala/org/scalajs/jsenv/AsyncJSEnv.scala

```diff
@@ -9,11 +9,47 @@
 
 package org.scalajs.jsenv
 
-import org.scalajs.core.tools.io._
-import org.scalajs.core.tools.classpath._
-import org.scalajs.core.tools.logging._
+import org.scalajs.core.tools.io.VirtualJSFile
+import org.scalajs.core.tools.jsdep.ResolvedJSDependency
+import org.scalajs.core.tools.linker.LinkingUnit
 
 trait AsyncJSEnv extends JSEnv {
-  def asyncRunner(classpath: CompleteClasspath, code: VirtualJSFile,
-      logger: Logger, console: JSConsole): AsyncJSRunner
+  def asyncRunner(libs: Seq[ResolvedJSDependency], sjsCode: VirtualJSFile,
+      code: VirtualJSFile): AsyncJSRunner
+
+  override def loadLibs(libs: Seq[ResolvedJSDependency],
+      sjsCode: VirtualJSFile): AsyncLibJSEnv = {
+    new AsyncLibJSEnv {
+      def unloadLibs: AsyncJSEnv = AsyncJSEnv.this
+
+      def jsRunner(code: VirtualJSFile): JSRunner =
+        AsyncJSEnv.this.jsRunner(libs, sjsCode, code)
+
+      def asyncRunner(code: VirtualJSFile): AsyncJSRunner =
+        AsyncJSEnv.this.asyncRunner(libs, sjsCode, code)
+    }
+  }
+}
+
+trait LinkingUnitAsyncJSEnv extends LinkingUnitJSEnv with AsyncJSEnv {
+  def asyncRunner(libs: Seq[ResolvedJSDependency], linkingUnit: LinkingUnit,
+      code: VirtualJSFile): AsyncJSRunner
```

Comment:
I just realized that there is a pretty big uglyness here: What symbols must be reached in this `linkingUnit`? The Rhino interpreter needs the same than the `BasicLinkerBackend`, since it uses the `Emitter` under the hood.
But this is not necessary at all...
One could, for example, imagine that we implement longs in Rhino by using real JVM longs. Of course the Scala.js project won't do so for testing, but someone else might...